app/controllers/application_controller.rb

```diff
@@ -72,7 +72,7 @@ def flash_and_redirect_back_with_message(exception)
       end
     end
 
-    redirect_to :back
+    redirect_back(fallback_location: (request.referer || root_path))
```

Comment:
Would love to gets some 👀 on this. This seems like the most reasonable action here.